lib/db/schema/patch-067-068.sql

```diff
@@ -0,0 +1,155 @@
+ALTER TABLE `unverifiedTokens`
+ADD COLUMN `shortCodeHash` BINARY(32) DEFAULT NULL,
+ALGORITHM = INPLACE, LOCK = NONE;
+
+CREATE INDEX `unverifiedTokens_shortCodeHash`
+ON `unverifiedTokens` (`shortCodeHash`)
+ALGORITHM = INPLACE LOCK=NONE;
+
+ALTER TABLE `unverifiedTokens`
+ADD COLUMN `shortCodeExpiresAt` BIGINT UNSIGNED DEFAULT NULL,
+ALGORITHM = INPLACE, LOCK = NONE;
+
+CREATE PROCEDURE `verifyTokenShortCode_1` (
+  IN `shortCodeHashArg` BINARY(32),
+  IN `uidArg` BINARY(16)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  SET @tokenVerificationId = NULL;
+  SELECT tokenVerificationId INTO @tokenVerificationId FROM unverifiedTokens WHERE uid = uidArg AND shortCodeHash = shortCodeHashArg;
+
+  IF @tokenVerificationId IS NOT NULL THEN
```

Comment:
What happens if `@tokenVerificationId` is `NULL` here?  IIUC control flow will fall through and execute the `UPDATE securityEvents` below, which will set `verified = true` for all security events where `tokenVerificationId` is `NULL`.  I think that's OK because such events will already have `verified=true` anyway, but it might be safer to bail out early if we don't find any matching `tokenVerificationId`.